development/templates/development/new_match_details.html

```diff
@@ -8,15 +8,62 @@ <h1>Match details</h1>
     {% if data %}
     <h1>Match details</h1>
     {% for log in data %}
-                <script>
-                addLogToArray(`{{log|safe}}`)
-                </script>
-            {% endfor %}
+        <script>
+        addLogToArray(`{{log|safe}}`)
+        </script>
+    {% endfor %}
     {% endif %}
     {% if not data %}
     <div>There are no logs to display</div>
     {% endif %}
     {% if data %}
+      <p>
+        <button class="btn btn-primary" type="button" data-toggle="collapse" data-target="#collapseExample" aria-expanded="false" aria-controls="collapseExample">
+          Download
+        </button>
+      </p>
+      <div class="collapse" id="collapseExample">
+        <div class="card card-body">
+          <div class="container">
+            <div class="row">
+              <div class="col">
+                <label for="valid_move">Validity</label>
+                  <select class="form-control" name="valid_move" id="valid_move">
+                    <optgroup label="Validity">
+                        {% for states in move_states %}
+                          {% if states != "gameover" %}
```

Comment:
why is this filtering happening here? shouldn't `move_states` be pre-filtered with the options we allow first?

Reply:
You're all right. I should have put this inside the `possible_states` property of the `FilterLogs` class. I'm changing it